### DISCUSSION TITLE
Cannot read properties of undefined (reading 'session') in next 13 app layout server component

### DISCUSSION THREAD

**Discussion by TommySorensen** [nextauthjs/next-auth#7635] [Help]
### Question 💬

It seems like i'm unable to read the session with the `getServerSession` function when my middleware is redirecting the user to the signin page. Anyone knows why the session is failing even though im reading it in a layout server component? 

### How to reproduce ☕️

I'm basically just trying to provide the SessionProvider with the session from the layout file. It works fine when visiting the frontpage, but when redirected to the login page, then it suddenly fails with an error:
<img width="809" alt="image" src="https://github.com/nextauthjs/next-auth/assets/4134473/75f8ef84-c7b1-4516-a43d-6fab5f6c19ef">

```jsx
// app/[locale]/layout.tsx
import { getServerSession } from 'next-auth/next';
import authOptions from '~/auth';
import { Session } from './session';

export default async function RootLayout({ children }: RootLayout) {
  const session = await getServerSession(authOptions);

  return (
    <Session session={session}>
      <html>
        {children}
      </html>
    </Session>
  );
}

// app/[locale]/session.tsx
'use client';

import { SessionProvider } from 'next-auth/react';
import { Session as SessionType } from 'next-auth';

export function Session({ session, children }: Session) {
  return <SessionProvider session={session}>{children}</SessionProvider>;
}

interface Session {
  session: SessionType | null;
  children: React.ReactNode;
}

```

My auth config is pretty basic. I'm using the "next-intl" package for localisation 
```js
...

const authMiddleware = withAuth(
`pages`.
  function onSuccess(req) {
    return intlMiddleware(req);
  },
  {
    callbacks: {
      authorized: ({ token }) => token != null,
    },
    pages: {
      error: '/auth/error',
      signIn: '/auth/signin',
      signOut: '/auth/signout',
    },
  }
);

...
```

This code is just a simplification of the issue.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by TommyLeong** [comment]
Hello @TommySorensen , did you manage to overcome this?

**Reply by TommySorensen** [reply]
Not really. I think i ended up avoiding the issue
